tutorials/ui/gui_using_fonts.rst

```diff
@@ -8,51 +8,698 @@ Godot allows you to set specific fonts for different UI nodes.
+
+Subpixel positioning can be adjusted. This is a `FreeType <https://freetype.org/>`__
+feature that allows glyphs to be rendered more closely to their intended form.
+The default setting of **Auto** automatically enables subpixel positioning at
+small sizes, but disables it at large font sizes to improve rasterization
+performance.
+
+You can force the subpixel positioning mode to **Disabled**, **One half of a
+pixel** or **One quarter of a pixel**. **One quarter of a pixel** provides the
+best quality, at the cost of longer rasterization times.
+
+Changing antialiasing, hinting and subpixel positioning has the most visible
+effect at smaller font sizes.
+
+.. _doc_using_fonts_mipmaps:
+
+Mipmaps
+^^^^^^^
+
+By default, fonts do not have mipmaps generated to reduce memory usage and speed
+up rasterization. However, this can cause downscaled fonts to become grainy. This
+can be especially noticeable with :ref:`doc_3d_text` that doesn't have **Fixed
+Size** enabled. This can also occur when displaying text with a traditional rasterized
+(non-:ref:`MSDF <doc_using_fonts_msdf>`) font in a Control node that has its scale
+lower than ``(1, 1)``.
+
+After selecting a font in the FileSystem dock, you can enable the **Mipmaps** in
+the Import dock to improve downscaled font rendering appearance.
+
+Mipmaps can be enabled on MSDF fonts as well. This can improve font rencering
+quality a little at smaller-than-default sizes, but MSDF fonts are already
+resistant to graininess out of the box.
+
+.. _doc_using_fonts_msdf:
+
+MSDF font rendering
+^^^^^^^^^^^^^^^^^^^
+
+Multi-channel signed distance field (MSDF) font rendering allows rendering fonts
+at any size, without having to re-rasterize them when their size changes.
+
+MSDF font rendering has 2 upsides over traditional font rasterization, which
+Godot uses by default:
+
+- The font will always look crisp, even at huge sizes.
+- There is less stuttering when rendering characters *at large font sizes* for
+  the first time, as there is no rasterization performed.
+
+The downsides of MSDF font rendering are:
+
+- Higher baseline cost for font rendering. This is usually not noticeable on
+  desktop platforms, but it can have an impact on low-end mobile devices.
+- Fonts at small sizes will not look as clear as rasterized fonts, due to the
+  lack of hinting.
+- Rendering new glyphs for the first time *at small font sizes* may be more
+  expensive compared to traditional rasterized fonts.
+  :ref:`doc_using_fonts_font_prerendering` can be used to alleviate this.
+- LCD subpixel optimization cannot be enabled for MSDF fonts.
+- Fonts with self-intersecting outlines will not render correctly in MSDF mode.
+  If you notice rendering issues on fonts downloaded from websites such as
+  `Google Fonts <https://fonts.google.com>`__, try downloading the font from the
+  font author's official website instead.
+
+Using emoji
+^^^^^^^^^^^
+
+Godot has full support for emoji, but the font used (or one of its
```

Comment:
`CBDT/CBLC` (embedded PNGs) and `SVG` emoji fonts are supported. 
`COLR/CPAL` emoji fonts (custom vector format) are not.